File: ExampleApp/app/src/main/java/com/motiv/example/Links.java
package com.motiv.example;

import android.widget.ImageView;
import androidx.annotation.NonNull;
import androidx.databinding.BindingAdapter;
import com.google.gson.*;
import com.google.gson.annotations.*;
import com.google.gson.reflect.*;
import com.squareup.picasso.Picasso;
import io.realm.*;
import java.util.*;
import java.util.concurrent.*;

public class Links extends RealmObject {

    private static final Gson gson = new Gson();
    @NonNull private java.lang.String id = UUID.randomUUID().toString();

    @SerializedName("edit")
    private com.motiv.example.Link edit;

    @SerializedName("self")
    private com.motiv.example.Link self;

    @SerializedName("avatar")
    private com.motiv.example.Link avatar;

    public java.lang.String getId() {

        return this.id;
    }

    public void setId(java.lang.String id) {
        this.id = id;
    }

    public com.motiv.example.Link getEdit() {

        return this.edit;
    }

    public void setEdit(com.motiv.example.Link edit) {
        this.edit = edit;
    }

    public com.motiv.example.Link getSelf() {

        return this.self;
    }

    public void setSelf(com.motiv.example.Link self) {
        this.self = self;
    }

    public com.motiv.example.Link getAvatar() {

        return this.avatar;
    }

    public void setAvatar(com.motiv.example.Link avatar) {
        this.avatar = avatar;
    }

    @BindingAdapter({"bind:imageUrl"})
    public static void loadImage(ImageView view, java.lang.String url) {

        Picasso.with(view.getContext()).load(url).into(view);
    }

    public static Links fromJson(String json) {
        return gson.fromJson(json, Links.class);
    }

    public static String toJson(Links object) {
        return gson.toJson(object, Links.class);
    }

    public static String toJsonArray(Links[] array) {
        return gson.toJson(array, Links[].class);
    }

    public static Links[] fromJsonArray(String array) {
        return gson.fromJson(array, Links[].class);
    }
}

File: ExampleApp/app/src/main/java/com/motiv/example/Links.kt
 
package com.motiv.example
import android.widget.ImageView
import androidx.annotation.NonNull
import androidx.databinding.BindingAdapter
import androidx.fragment.app.*
import com.bumptech.glide.Glide
import com.google.gson.*
import com.google.gson.annotations.*
import com.google.gson.reflect.*
import dagger.*
import dagger.android.*
import dagger.android.support.*
import io.realm.*
import java.util.*
import java.util.concurrent.*
import javax.inject.*

public open class Links : RealmObject() {

    @NonNull private var id: String = UUID.randomUUID().toString()

    @SerializedName("edit")

    private var edit: com.motiv.example.Link = com.motiv.example.Link()

    @SerializedName("self")

    private var self: com.motiv.example.Link = com.motiv.example.Link()

    @SerializedName("avatar")

    private var avatar: com.motiv.example.Link = com.motiv.example.Link()

    fun getId(): String {
        return this.id
    } fun setId(id: String) {
        this.id = id
    } fun getEdit(): com.motiv.example.Link {
        return this.edit
    } fun setEdit(edit: com.motiv.example.Link) {
        this.edit = edit
    } fun getSelf(): com.motiv.example.Link {
        return this.self
    } fun setSelf(self: com.motiv.example.Link) {
        this.self = self
    } fun getAvatar(): com.motiv.example.Link {
        return this.avatar
    } fun setAvatar(avatar: com.motiv.example.Link) {
        this.avatar = avatar
    } @BindingAdapter("bind:imageUrl")
    fun loadImage(view: ImageView, url: String) {
        Glide.with(view.getContext())
            .load(url)
            .into(view)
    } companion object {
        val gson: Gson = Gson()
        fun fromJson(json: String): Links {
            return gson.fromJson(json, Links::class.java)
        }

        fun toJson(obj: Links): String {
            return gson.toJson(obj, Links::class.java)
        }

        fun toJsonArray(obj: Links): String {
            return gson.toJson(obj, Array<Links>::class.java)
        }

        fun fromJsonArray(json: String): Array<Links> {
            return gson.fromJson(json, Array<Links>::class.java)
        }
    }
}
